test/framework/clusterctl/client.go

```diff
@@ -90,23 +96,35 @@ func Init(_ context.Context, input InitInput) {
 
 // InitWithBinary uses clusterctl binary to run init with the list of providers defined in the local repository.
 func InitWithBinary(_ context.Context, binary string, input InitInput) {
-	log.Logf("clusterctl init --core %s --bootstrap %s --control-plane %s --infrastructure %s --config %s --kubeconfig %s",
+	log.Logf("clusterctl init --core %s --bootstrap %s --control-plane %s --infrastructure %s --ipam %s --extension %s --config %s --kubeconfig %s",
 		input.CoreProvider,
 		strings.Join(input.BootstrapProviders, ","),
 		strings.Join(input.ControlPlaneProviders, ","),
 		strings.Join(input.InfrastructureProviders, ","),
+		strings.Join(input.IPAMProviders, ","),
+		strings.Join(input.RuntimeExtensionProviders, ","),
 		input.ClusterctlConfigPath,
 		input.KubeconfigPath,
 	)
 
-	cmd := exec.Command(binary, "init", //nolint:gosec // We don't care about command injection here.
-		"--core", input.CoreProvider,
-		"--bootstrap", strings.Join(input.BootstrapProviders, ","),
-		"--control-plane", strings.Join(input.ControlPlaneProviders, ","),
-		"--infrastructure", strings.Join(input.InfrastructureProviders, ","),
-		"--config", input.ClusterctlConfigPath,
-		"--kubeconfig", input.KubeconfigPath,
-	)
+	args := []string{binary, "init", "--config", input.ClusterctlConfigPath, "--kubeconfig", input.KubeconfigPath}
```

Comment:
Q: Do we have other funcs in this file where we need a similar handling?

Reply:
AFAIK this is the only place where we are passing flags to the binary, and we should deal with an older version of the binary not supporting those flags.
Instead, when using clusterctl as a library we are always using the latest version of the code and this should not be a problem

Reply:
Sounds good, thx!